home2.2/Main.java

```diff
@@ -0,0 +1,190 @@
+            int row = 0;
+            while ((line = reader.readLine()) != null && row < rows) {
+                String[] values = line.split(" ");
+                if (values.length != cols) {
+                    System.err.println("Invalid number of pixels in row " + (row + 1));
+                    System.exit(1);
+                }
+                for (int col = 0; col < cols; col++) {
+                    String[] rgb = values[col].split(",");
+                    if (rgb.length != 3) {
+                        System.err.println("Invalid pixel format in row " + (row + 1) + ", column " + (col + 1));
+                        System.exit(1);
+                    }
+                    int r = Integer.parseInt(rgb[0]);
+                    int g = Integer.parseInt(rgb[1]);
+                    int b = Integer.parseInt(rgb[2]);
+                    if (!isValidColor(r, g, b)) {
+                        System.err.println("Pixel color out of range in row " + (row + 1) + ", column " + (col + 1));
+                        System.exit(1);
+                    }
+                    image[row][col] = new Pixel(r, g, b);
+                }
+                row++;
+            }
+            if (row < rows) {
+                System.err.println("Not enough rows in the input image");
+                System.exit(1);
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    public void writeImage(String filename) {
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(filename));
+            for (int i = 0; i < rows; i++) {
+                for (int j = 0; j < cols; j++) {
+                    writer.write(image[i][j].r + "," + image[i][j].g + "," + image[i][j].b + " ");
+                }
+                writer.newLine();
+            }
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    public void processAndSave(String outputFilename) {
+        if (favorite == null) {
+            System.err.println("Favorite color not set");
+            System.exit(1);
+        }
+        // Check if unfavorite color is provided and if it is present in the image
+        if (unfavorite != null) {
+            boolean unfavColorFound = false;
+            for (Pixel[] row : image) {
+                for (Pixel pixel : row) {
+                    if (pixel != null && pixel.r == unfavorite.r && pixel.g == unfavorite.g && pixel.b == unfavorite.b) {
+                        unfavColorFound = true;
+                        break;
+                    }
+                }
+                if (unfavColorFound) {
+                    break;
+                }
+            }
+            if (!unfavColorFound) {
+                System.err.println("Unfavorite color not found in the image");
+                System.exit(1);
```

Comment:
using direct exit call inside some class may not be the best idea. Consider throwing an exception therefore top layers of abstractions can decide what to do with such a situation - terminate program, retry some logic, etc.